src/renderer/pages/Project.tsx

```diff
@@ -1,28 +1,80 @@
-import { Stack } from '@mui/material';
-import { Box } from '@mui/system';
+import { Box, Stack } from '@mui/material';
+import { useCallback, useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import StandardButton from 'renderer/components/StandardButton';
-import { projectOpened } from 'renderer/store/actions';
+import TranscriptionBlock from 'renderer/components/TranscriptionBlock';
+import { transcriptionCreated } from 'renderer/store/actions';
+import { Transcription } from 'sharedTypes';
 import { ApplicationStore } from '../store/helpers';
 
+/*
+This is the page that gets displayed while you are editing a video.
+It will be primarily composed of the transcription area, an editable text box whose
+changes get reflected in the video. In addition to that, there is a video preview
+section to the side among other things.
+*/
 const ProjectPage = () => {
+  const dispatch = useDispatch();
   const currentProject = useSelector(
     (store: ApplicationStore) => store.currentProject
   );
 
-  const dispatch = useDispatch();
+  // RK: I really shouldn't use transcriptionCreated here - but i'm lazy and it works
+  const saveTranscription: (transcription: Transcription) => void = useCallback(
+    (transcription) => dispatch(transcriptionCreated(transcription)),
+    [dispatch]
+  );
+
+  const deleteWord = (firstWordIndex: number, numberOfWords: number) => {
+    if (currentProject && currentProject.transcription) {
+      // eslint-disable-next-line no-plusplus
+      for (let i = firstWordIndex; i < firstWordIndex + numberOfWords; i++) {
+        currentProject.transcription.words[i].deleted = true;
```

Comment:
If you want these actions to be undoable, then dispatch them as ops with 'do' and 'undo' actions (see: [undo stack doc](https://docs.google.com/document/d/1fBLBj_I3Y4AgRnIHzJ-grsXvzoKUBA03KNRv3DzABAg/edit), [video if you're up for it](https://www.youtube.com/watch?v=D9laah5bcio)

Especially don't mutate state - this isn't how redux (or react in general) is supposed to work

Happy to run you through this if you like

Reply:
@rileykeane since you're the PM for this feature I'll let you make the call, but if I remember correctly undo is out of scope for this sprint.

Reply:
Sure, it's more about conforming to the existing approach for reversible actions, but I also realise that standard was only implemented a few days ago and so it's fine if you want to merge this as is and then update later.

Reply:
undo/redo was not part of the MVP. However since it seems a big chunk of the heavy lifting to implement this is already done, we can potentially look into it next sprint. Of course need to prioritise actual MVP features over this.

Reply:
Yeah justification for it here: https://docs.google.com/document/d/1fBLBj_I3Y4AgRnIHzJ-grsXvzoKUBA03KNRv3DzABAg/edit#heading=h.7c9bao1d81db

On your end all that's needed is do + undo redux actions wrapped in an op, then dispatch that with dispatchOp. All the actual undo functionality is abstracted away

Reply:
I do understand that it wasn't originally part of the mvp - so defs no need to do it in this PR. All I'm saying is in terms of "make it so the user can undo" we're already 90% of the way there, you just have to provide a reversible action